pkg/fftypes/subscription.go

```diff
@@ -26,13 +26,28 @@ import (
 
 // SubscriptionFilter contains regular expressions to match against events. All must match for an event to be dispatched to a subscription
 type SubscriptionFilter struct {
-	Events string `json:"events,omitempty"`
+	Events          string                `json:"events,omitempty"`
+	Message         MessageFilter         `json:"message,omitempty"`
+	Transaction     TransactionFilter     `json:"transaction,omitempty"`
+	BlockchainEvent BlockchainEventFilter `json:"blockchainevent,omitempty"`
```

Comment:
See earlier comments about being able to parse in the old options at the top level:

```
	DeprecatedTopics string `json:"topics,omitempty"`
	DeprecatedTag    string `json:"tag,omitempty"`
	DeprecatedGroup  string `json:"group,omitempty"`
	DeprecatedAuthor string `json:"author,omitempty"`
```